analyzers/tests/SonarAnalyzer.Test/TestCases/SymbolicExecution/Roslyn/NullPointerDereference.cs

```diff
@@ -2161,3 +2161,36 @@ private void Foo(List<string> values)
         }
     }
 }
+
+public class Repro_NoCollectionInstanceOnFirstOrDefault
+{
+    public IQueryable<AClass> Elements { get; set; }
+    public IQueryable<AStruct> StructElements { get; set; }
+
+    public void Put(long id, AClass model)
+    {
+        var element = Elements.Where(x => x.Name == "name").FirstOrDefault(x => x.Id == id);
+        var newElement = new AClass()
+        {
+            Name = element.Name, // Noncompliant
+        };
```

Comment:
This can be simplified
```suggestion
        _ = element.Name; // Noncompliant
```